feat: LSDV-5522: Add Heidi tips to general project settings

juliosgarbi opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [x] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Describe the reason for change
Add Heidi tips to general project settings also add workspace placeholder



#### What does this fix?
_(if this is a bug fix)_



#### What is the new behavior?
_(if this is a breaking or feature change)_



#### What is the current behavior?
_(if this is a breaking or feature change)_



#### What libraries were added/updated?
_(list all with version changes)_



#### Does this change affect performance?
_(if so describe the impacts positive or negative)_



#### Does this change affect security?
_(if so describe the impacts positive or negative)_



#### What alternative approaches were there?
_(briefly list any if applicable)_



#### What feature flags were used to cover this change?
_(briefly list any if applicable)_



### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit



### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

nick-skriabin committed:
Introduce Heidi tips

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5901625542

nick-skriabin committed:
Fix random tip selection

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5902201016

nick-skriabin committed:
Cookie to dismiss tips

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5903825521

nick-skriabin committed:
Caption component

nick-skriabin committed:
Fix caption styles

nick-skriabin committed:
Use caption in project creation

juliosgarbi committed:
feat: LSDV-5522: Add Heidi tips to general project settings

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5904073950

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5904090485

nick-skriabin committed:
Option to add URL params to the links

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5904701057

nick-skriabin committed:
Add feature flag

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5904791829

juliosgarbi commented:
Nick already did it on LEAP-13

juliosgarbi commented:
/git merge develop
> Successfully pushed new changes:
> Merge remote-tracking branch 'origin/develop' into fb-lsdv-5522 (ddb1b6e6bbe1c8f949e36fd82b0e1458411b28c0)
>
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/5905370048)

juliosgarbi committed:
Merge remote-tracking branch 'origin/develop' into fb-lsdv-5522

juliosgarbi committed:
merge with main branch

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5905425820

juliosgarbi committed:
fix links

juliosgarbi committed:
change single quote to double quote

juliosgarbi committed:
merge conflicts

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5905727264

juliosgarbi committed:
fix some lint issues

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5907055489

juliosgarbi committed:
add new GA params

juliosgarbi committed:
add new GA params

juliosgarbi committed:
move the code to be defined before HeidiTip

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5907116794

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5907139364

juliosgarbi committed:
add link to workspace dropdown

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5907221213

nick-skriabin committed:
Fix typo

nick-skriabin committed:
Fix link behavior

nick-skriabin committed:
Fix link behavior

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5925171669

nick-skriabin committed:
Add general settings tip link params

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5925311371

nick-skriabin committed:
Add server_id and user_id to createURL

nick-skriabin committed:
Add server_id and user_id to createURL

nick-skriabin committed:
Remove unnecessary server_id from content

nick-skriabin committed:
Remove unnecessary server and user ids from project creation tip

nick-skriabin committed:
Remove unnecessary server and user ids from project creation tip

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5925352703

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5925378122

nick-skriabin committed:
Remove unnecessary user and server IDs

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5925597877

nick-skriabin committed:
Merge branch 'fb-lsdv-5520/project-creation-tip' into fb-lsdv-5522

nick-skriabin committed:
Fix link param type

nick-skriabin committed:
Fix link params

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5926057613

nick-skriabin committed:
Support SERVER_ID

nick-skriabin committed:
Remove placeholder

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5926637110

nick-skriabin committed:
Move server_id assignment

hlomzik commented on label_studio/frontend/src/components/HeidiTips/HeidiTips.tsx:
```diff
@@ -0,0 +1,18 @@
+import { dismissTip, getRandomTip, isTipDismissed } from "./utils";
+
+export const HeidiTips: FC<HeidiTipsProps> = memo(({
+  collection,
```
Add a type here, otherwise it's unclear how to use this component and what the collection is. Overall I'd make usage more clear, for example, collections keys can form a Enum, which can be used here as type and in external component as parameter like `<HeidiTips collection={TIPS.GENERAL} />`

hlomzik commented on label_studio/frontend/src/components/HeidiTips/HeidiTip.tsx:
```diff
@@ -0,0 +1,64 @@
+import { Tooltip } from "../Tooltip/Tooltip";
+import { createURL } from "./utils";
+
+const HeidiLink: FC<{ link: Exclude<Tip["link"], undefined> }> = ({
```
`link` should be `Tip["link"]` to be correct and still have a connection to general `Tip` type

juliosgarbi commented:
/git merge develop
> **Error**: failed to execute "merge" command
>
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/5940008309)

juliosgarbi committed:
merge conflcits

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5940095387

juliosgarbi committed:
fix styles and links from general settings

juliosgarbi committed:
fix styles and links from general settings

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5940720736

yyassi-heartex commented on label_studio/frontend/src/pages/Settings/GeneralSettings.js:
```diff
@@ -24,60 +29,95 @@ export const GeneralSettings = () => {
-            labelProps={{large: true}}
-          />
+    <Block name="general-settings">
+      <div style={{ width: 488 }}>
```
since BEM is being used this would be a good opportunity to update this native div to an Elem (if its still needed) ad update the css so we dont add inline styles here unless you feel this is too big a lift for this ticket and it should be addressed in another ticket

juliosgarbi replied:
You're right @yyassi-heartex, it's fixed, thanks!

juliosgarbi committed:
change div to Elem

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5941264161

juliosgarbi committed:
add missing URL

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/5943299142
